yt_dlp/extractor/ant1newsgr.py

```diff
@@ -0,0 +1,184 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+from urllib.parse import (
+    parse_qs,
+    urlparse,
+    urlunparse,
+)
+
+from .common import InfoExtractor
+from ..utils import (
+    HEADRequest,
+    ExtractorError,
+    determine_ext,
+    smuggle_url,
+    unsmuggle_url,
+    unescapeHTML,
+)
+
+
+class Ant1NewsGrBaseIE(InfoExtractor):
+    @staticmethod
+    def _smuggle_parent_info(url, **info_dict):
+        return smuggle_url(url, {'parent_info': info_dict})
+
+    @staticmethod
+    def _unsmuggle_parent_info(url):
+        unsmuggled_url, data = unsmuggle_url(url, default={'parent_info': {}})
+        return unsmuggled_url, data['parent_info']
```

Comment:
why? Instead just use `'_type': 'url_transparent'` on the urls returned by `_extract_urls`

Reply:
I am afraid I do not understand your point.

These functions (de)serialize the parent_info kwargs provided to `Ant1NewsGrEmbedIE._extract_urls`, namely metadata from the embedding page.  The latter function is used both by the generic extractor and `Ant1NewsGrArticleIE`, which in addition extracts JSON-LD metadata. Since the particular functions are only used once they could be in-lined, however this code is generic and it looked cleaner this way. However I do not see how returning `{'_type': 'url_transparent'}` changes anything so that such metadata from the embedding page turn up and can be used for the embedded video?

Reply:
I think I finally got your point, and thus tried to switch from `(un)smuggle_url` to using `url_transparent` in the last commits (af08027 for some changes in `url_result` and `playlist_from_matches`, and bb12bda).
Is this what you meant?

A perhaps important difference with this approach is that the source of the metadata (parent) gets to dictate what fields it overrides in the subsequently resolved extraction.

Reply:
Yes

> A perhaps important difference with this approach is that the source of the metadata (parent) gets to dictate what fields it overrides in the subsequently resolved extraction.

Is that a problem?

Reply:
> Is that a problem?

Not that I can think of. It did work out in this case.

Reply:
After rebasing and fixing merge conflicts (c7b91ddea0781abf1cc4a9550e9dd4b0713e57fe was replaced by 311b6615d85d3530f2709c50e4223ff3b6b14361) I realized I still need a way to call `playlist_from_matches` and have it set `url_transparent` and override some fields. So I added `video_kwargs` to it and also added `url_result_or_playlist_from_matches`, a wrapper that either calls `url_result` or the former method.

Reply:
> After rebasing and fixing merge conflicts (c7b91dd was replaced by 311b661) I realized I still need a way to call playlist_from_matches and have it set url_transparent and override some fields. So I added video_kwargs to it

I intentionally did not add such a key because it does not make sense to do, as we discussed in the other PR. It makes sense to propagate the title only when there is a single entry in the page. If there are multiple embeds, we do not want all titles to end up being the same as the playlist

Reply:
I am also overriding the timestamp here.